packages/@aws-cdk/core/lib/deps.ts

```diff
@@ -70,31 +120,67 @@ export function addDependency<T extends Element>(source: T, target: T, reason?:
   // `source` is a direct or indirect nested stack of `target`, and this is not
   // possible (nested stacks cannot depend on their parents).
   if (commonStack === target) {
-    throw new Error(`Nested stack '${sourceStack.node.path}' cannot depend on a parent stack '${targetStack.node.path}': ${reason}`);
+    throw new Error(`Nested stack '${sourceStack.node.path}' cannot depend on a parent stack '${targetStack.node.path}': ${reasonDescription}`);
   }
 
   // we have a common stack from which we can reach both `source` and `target`
   // now we need to find two resources which are defined directly in this stack
   // and which can "lead us" to the source/target.
-  const sourceResource = resourceInCommonStackFor(source);
-  const targetResource = resourceInCommonStackFor(target);
-  sourceResource._addResourceDependency(targetResource);
-
-  function resourceInCommonStackFor(element: CfnResource | Stack): CfnResource {
-    const resource = Stack.isStack(element) ? element.nestedStackResource : element;
-    if (!resource) {
-      throw new Error('assertion failure'); // see "assertion" above
+  const sourceResource = resourceInCommonStackFor(source, commonStack);
+  const targetResource = resourceInCommonStackFor(target, commonStack);
+  switch (operation) {
+    case DependencyOperation.ADD: {
+      sourceResource._addResourceDependency(targetResource);
+      break;
+    }
+    case DependencyOperation.REMOVE: {
+      sourceResource._removeResourceDependency(targetResource);
+      break;
     }
+    default: {
+      throw new Error(`Unsupported dependency operation: ${operation}`);
+    }
+  }
+}
 
-    const resourceStack = Stack.of(resource);
+/**
+ * Get a list of all resource-to-resource dependencies assembled from this Element, Stack or assembly-dependencies
+ * @param source The source resource/stack (the dependent)
+ */
+export function obtainDependencies(source: Element) {
```

Comment:
I think the verb here and elsewhere should be `get` instead of `obtain`

Reply:
I think jsii yelled at me for calling things `get*` so I started using `obtain` when I saw it used elsewhere.

Reply:
Ah ok yeah, here's the error. This means we can't use `get*` method names at all, doesn't it?
```error JSII5000: Methods and properties cannot have names like "getXxx": those conflict with Java property getters. Rename "CfnResource.getDependencies"```